### ISSUE TITLE
[a11y]: CreateFullPage headings are not semantically correct

### ISSUE THREAD

**Issue by shocquu** [carbon-design-system/ibm-products#6815]
### Package

@carbon/ibm-products

### Browser

Chrome, Safari, Firefox, Edge

### Package version

v2.57.1

### React version

v18.0.0

### Automated testing tool and ruleset

Google Lighthouse, IBM Equal Access Accessibility Checker, Storybook

### Description

The headings in **CreateFullPage** do not follow the proper heading sequence; they skip from `h1` directly to `h5`. Ideally, we would prefer to manage the starting order, especially when Success Criterion 2.4.6: Headings and Labels is integrated into a modal, where the headings should commence from `h2`.

### WCAG 2.1 Violation

Success Criterion 2.4.6: Headings and Labels

### Reproduction/example

https://ibm-products.carbondesignsystem.com/iframe.html?globals=viewport%3Abasic&args=&id=ibm-products-patterns-create-flows-createfullpage--create-full-page-with-global-header&viewMode=story

### Steps to reproduce

1. Import \<CreateFullPage/> component,
2. Run the accessibility checker (Lighthouse, IBM Equal Access Accessibility Checker, etc.)
3. Get the `Heading levels should only increase by one` violation

![Image](https://github.com/user-attachments/assets/8538608a-c21d-4ab0-8e3e-5cbd91e6dd35)

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
